hs/src/Reach/AST/DLBase.hs

```diff
@@ -755,11 +758,14 @@ data DLRemoteALGO = DLRemoteALGO
   , ralgo_onCompletion :: DLRemoteALGOOC
   , ralgo_strictPay :: Bool
   , ralgo_rawCall :: Bool
+  , ralgo_simValues :: (DLArg, -- netRecv
+                        Either (Maybe [DLArg]) DLArg, -- nonNetRecv (Left = UInt bill amounts given by user, Right = compiled into Tuple of UInts)
+                        DLArg) -- returnVal
```

Comment:
Small issue... I think you should make a record for this. I find it very tedious to work with tuples in all the different passes. Furthermore, I think it would be nice if we don't enforce them to be present and not all of them. So I really think it should be...

```suggestion
  , ralgo_simValues :: (DLArg, -- netRecv
                        Either (Maybe [DLArg]) DLArg, -- nonNetRecv (Left = UInt bill amounts given by user, Right = compiled into Tuple of UInts)
                        DLArg) -- returnVal
  , ralgo_sim_netRecv :: Maybe DLArg
  , ralgo_sim_bill :: Maybe (Either [DLArg] DLArg) -- not sure about this
  , ralgo_sim_ret :: Maybe DLArg
```

Maybe `netRecv` and `bill` don't need to be `Maybe` and the `0` can be put in by the code generator, but there's no way to represent the `undefined` for `ret`